### PR TITLE
API 2826 Questionnaire Response Read

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

File: patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java [hunk 1]
```diff
@@ -112,6 +113,7 @@ private static void populate(@NonNull Db db) {
   @SneakyThrows
   private static void questionnaire(@NonNull Connection connection) {
     Set<String> ids = new HashSet<>();
+    String sql = sql("app.questionnaire", List.of("id", "payload", "version"));
```
Comment (jblefkove-va) [inline patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java:116]: This line is a from a merge conflict. (See line 128.) 

There don't appear to be any intended changes in this file, so overwrite it with the version from master.

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseEntity.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+@Data
+@Entity
+@Builder
+@Table(name = "questionnaireresponse", schema = "app")
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseEntity.java:20]: camel-case the table name

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@RequestMapping(
+    value = "/r4/QuestionnaireResponse",
+    produces = {"application/json", "application/fhir+json"})
+@AllArgsConstructor(onConstructor_ = @Autowired)
+public class QuestionnaireResponseController {
+
+  private final QuestionnaireResponseRepository repository;
+
+  @SneakyThrows
+  static <T> T deserializedPayload(
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Move this method to a utility class so it can be shared between controllers

Comment (that-guy-is-it) [reply patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Any preference as to what that class should be called?

Comment (jblefkove-va) [reply patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: `Controllers`?

Comment (that-guy-is-it) [reply patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Sure, sounds good to me.

Comment (apeirats-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Let's not duplicate this everywhere. Move to a static util class of some sort, like `SerializationUtils`

Comment (that-guy-is-it) [reply patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Yes, Peter said this as well, I am on it right now. I also am gonna make tests for it, to not duplicate the test either.

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseEntity.java (hunk 2 above)
Comment (apeirats-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseEntity.java:20]: May not matter but actual name is `QuestionnaireResponse`

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java (hunk 3 above)
Comment (apeirats-va) [reply patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/QuestionnaireResponseController.java:37]: Guess I should've refreshed the page before adding comments

**Review by jblefkove-va — APPROVED** [review]
